pkg/machine/qemu.go

```diff
@@ -99,13 +104,57 @@ func (q *QEMU) Config() types.MachineConfig {
 	return q.machineConfig
 }
 
+// qemu monitor: https://qemu-project.gitlab.io/qemu/system/monitor.html
+// nice explanation of how it works: https://unix.stackexchange.com/a/476617
+// unix sockets with golang: https://dev.to/douglasmakey/understanding-unix-domain-sockets-in-golang-32n8
+func (q *QEMU) Screenshot() (string, error) {
+	conn, err := net.Dial("unix", q.monitorSockFile())
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close()
+
+	// Create a temp file name
+	f, err := os.CreateTemp("", "qemu-screenshot-*.png")
+	if err != nil {
+		return "", err
+	}
+	f.Close()
+	os.Remove(f.Name())
+
+	cmd := fmt.Sprintf("screendump %s\r\n", f.Name())
+	n, err := fmt.Fprint(conn, cmd)
+	if err != nil {
+		return "", err
+	}
+
+	if n != len(cmd) {
+		return "", fmt.Errorf("didn't send the full command (%d out of %d bytes)", n, len(cmd))
+	}
+
+	// If there is nothing for more than a second, stop
+	if err := conn.SetReadDeadline(time.Now().Add(time.Second)); err != nil {
+		return "", err
+	}
+	for {
+		b := make([]byte, 1024)
+		if _, err := conn.Read(b); err != nil {
+			break
+		}
+	}
```

Comment:
It seems that the `screendump image.png` command doesn't have any effect until we read the data from the socket. I would expect reading the data to be irrelevant but after trial and errors, this seems to be necessary for some reason.

Reply:
Put it in a comment